client/my-sites/checkout/checkout/index.jsx

```diff
@@ -414,14 +414,17 @@ export class Checkout extends React.Component {
 			// We cannot simply compare `hostname` to `selectedSiteSlug`, since the latter
 			// might contain a path in the case of Jetpack subdirectory installs.
 			if ( adminUrl && redirectTo.startsWith( `${ adminUrl }post.php?` ) ) {
+				const plan = get( query, [ 'plan' ] );
+				const action = get( query, [ 'action' ] );
 				const sanitizedRedirectTo = formatUrl( {
 					protocol,
 					hostname,
 					port,
 					pathname,
 					query: {
 						post: parseInt( get( query, [ 'post' ] ), 10 ),
-						action: 'edit',
+						...( action && { action } ),
```

Comment:
Hard-wiring `action` to `'edit'` is actually intended. It's my way of sanitizing this -- to narrow it down to what we know is allowed and harmless. My thinking is that we can allow other actions (probably based on a whitelist though) in case the need should ever arise.

(To clarify, this part of the code is like ten days old -- see #35027.)

Reply:
Gotcha - will revert this back and hardcode `plan` (`plan_upgraded`) to `1` I presume also?

Reply:
Sounds good :+1: